### PR TITLE
Fix: randomize unit test fixture names to avoid concurrency issues

### PR DESCRIPTION
**Pull request by georgesittas**
Addresses the 9th item in https://github.com/TobikoData/sqlmesh/issues/1637

### CODE REVIEW

**Review by eakmanrq — COMMENTED** [review]
Does the unique id generated for the test get surfaced back to the user in some way? I also wonder if we could come up with something more meaningful to the user as a way of differentiating tests. Maybe give the whole test run a unique ID that is shown to the user and then all fixtures are suffixes with that unique id?

File: sqlmesh/core/test/definition.py [hunk 1]
```diff
@@ -121,9 +123,7 @@ def tearDown(self) -> None:
         """Drop all fixture tables."""
         for table in self.body.get("inputs", {}):
             self.engine_adapter.drop_view(
-                _fully_qualified_test_fixture_name(table, self.dialect)
-                if table in self.models
-                else table
+                self._test_fixture_table(table).sql() if table in self.models else table
```
[the remark this answers is not in the document]
Comment (georgesittas) [reply sqlmesh/core/test/definition.py:126]: That's what we used to do using `_fully_qualified_test_fixture_name`. I don't remember the context around that, so I just wanted to preserve the behavior.

[the remark this answers is not in the document]
Comment (georgesittas) [reply sqlmesh/core/test/definition.py:126]: > doesn't drop_view take in a table?

Ah I see, yeah I think we can get rid of the `sql` call. Making the change.

**Comment by georgesittas** [comment]
> Does the unique id generated for the test get surfaced back to the user in some way?

Hm, let me check. I'm not sure if it gets surfaced during error failure. Perhaps this would be exposed to users that want to query the testing db directly (see "Add support for "dumping" duckdb state on test failure to speed up debugging").

> Maybe give the whole test run a unique ID that is shown to the user and then all fixtures are suffixes with that unique id?

Ah, that makes sense. I can refactor this approach to associated a test run with a random ID like you're suggesting.

**Comment by georgesittas** [comment]
Done, updated a unit test to check the random ID suffix as well.

**Review by tobymao — APPROVED** [review]

**Pull request merged**